File: Excel/500-599/566/566 Challenge.py
import pandas as pd

path = "566 Count in Columns.xlsx"

input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=12)
test = pd.read_excel(path, usecols="E:J", skiprows=1, nrows=4).fillna('')

result = (input.melt(var_name='basket', value_name='fruit')
          .dropna()
          .groupby(['fruit', 'basket'])
          .size()
          .reset_index(name='Count')
          .pivot_table(index='fruit', columns='Count', values='basket', aggfunc=lambda x: ', '.join(sorted(x)))
          .reset_index()
          .rename_axis(None, axis=1)
          .sort_values(by='fruit')
          .reindex(columns=['fruit', 1, 2, 3, 4, 5])
          .fillna('')
          .rename(columns={'fruit': 'Count'}))

print(result.equals(test)) # True

File: Excel/500-599/566/566 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/566 Count in Columns.xlsx"
input = read_excel(path, range = "A2:C14")
test  = read_excel(path, range = "E2:J6")

result = input %>%
  pivot_longer(everything(), names_to = "basket", values_to = "fruit") %>%
  summarise(Count = n(), .by = c(fruit, basket)) %>%
  na.omit() %>%
  pivot_wider(names_from = Count,
              values_from = basket, 
              values_fn = list(basket = ~ str_c(sort(.x), collapse = ", "))) %>%
  arrange(fruit) %>%
  select(Count = fruit,`1`, `2`, `3`, `4`, `5`) 

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE
